src/cascadia/TerminalSettingsModel/IInheritable.h

```diff
@@ -0,0 +1,156 @@
+/*++
+Copyright (c) Microsoft Corporation
+Licensed under the MIT license.
+
+Module Name:
+- IInheritable.h
+
+Abstract:
+- An interface allowing settings objects to inherit settings from a parent
+
+Author(s):
+- Carlos Zamora - October 2020
+
+--*/
+#pragma once
+
+namespace winrt::Microsoft::Terminal::Settings::Model::implementation
+{
+    template<typename T>
+    struct IInheritable
+    {
+    public:
+        T* CreateChild() const

```

Comment:
I'd rather you return a `com_ptr` from this; otherwise, there's a brief period where nobody owns the pointer and it can leak.